### PR TITLE
STN-607: Implementation of Testimonial Paragraph/UI Pattern for views

### PR DESCRIPTION
**Pull request by heatherdesigns**
# [STN-607](https://sparkbox.atlassian.net/browse/STN-607) READY FOR REVIEW

## Summary
Creates a new Testimonial Paragraph and UI Pattern to be used in views . Used on both Colorful and Traditional. The styles are different to complement each theme. 

Design comps for testimonial: [Figma](https://www.figma.com/file/RiJgL9DltLHzig4ynM34iU/H-and-S-Testimonial-Block-Quote?node-id=0%3A1)

## Need Review By (Date)
asap

## Urgency
high

## Steps to Test
1. In your CLI, run `npm run test` and confirm our tests pass.
2. In order for the testimonial pattern to be recognized, you will need to clear your cache by running `lando drush @sparkbox_sandbox.local cr`.
3. To test a paragraph type, create a new flexible page or open an existing one. Under `Components` add a `Testimonial`. Fill in all the fields. 
4. To test the testimonial as a UI pattern in a view, add a view to a flexible page. Under  `Components` add a `View`. I chose to add `Custom Econ Events` with a Display of `Past Events.`
    
    add view
<br>
    Edit the view to use the testimonial UI pattern. Under `Format > Show` click on `Settings`.
<br>
    veiw settings
<br>
    Edit the Row style options. Set `Pattern` to `Testimonial`. Leave the `Variant` to `Top (Default)`. Set `Content: Title` to `Position`, `Content: Image` to `Image`, `Content: Body` to `Quote`, `Content: Speaker` to `Name`, and `Content: Link to Content` to `Link`. All the other sources should be set to `- Hidden -`. Apply and save the view.
<br>

row style options

5. Test top / bottom quote options
6. Use the Testing Checklist below as a reference for all the things you should consider testing.

### Themes
- [x] Colorful (Note that the image on the Colorful theme has a border-radius that crops it into a circle, while the image in the Traditional theme is a square.)
- [x] Traditional

### Theme Settings
- [x] Dark Color Inversion

### Utility Classes
Layout Builder
- [x] hb-dark-pattern
- [x] hb-blockquote-black-text (When added to layout builder, this class should make the color of the quote text black.)

#### Views
- [x] hb-dark-pattern

#### Fields --> The utility classes work but I don't recommend using them on the testimonial.
- [x] hb-title
- [x] hb-highlighted-label
- [x] hb-subtitle
- [x] hb-divider

### Component considerations
- [x] Rows
- [x] Wells

### Content considerations
- [x] Rows with single and double wide content
- [x] Longer names / positions
- [x] Examples without all the fields used & different combinations of fields used.
- [x] Text area / Testimonial / Text area

## Screenshot

sparkbox-sandbox suhumsci loc_qa_testimonial-paragraphui-pattern-views

### CODE REVIEW

File: docroot/themes/humsci/humsci_basic/src/scss/tools/_functions.icons.scss [hunk 1]
```diff
@@ -37,6 +37,6 @@
   } @else if ($icon == 'ellipse') {
     @return '<svg viewBox="0 0 11 11" fill="none"><circle cx="5.49989" cy="5.49994" r="5.5" fill="#{$color}"/></svg>';
   } @else if ($icon == 'quote-mark') {
-    @return '<svg viewBox="0 0 88 48" fill="none"><path d="M28.2,47.1c5.4,0,9.3-3.7,9.3-9.6c0-5.8-4.5-8.9-9.3-8.9c-0.5,0-1.1,0-1.6,0.2C26.7,18.9,31,12.8,40,7l-4.2-6.8	c-12.1,7-18.5,17.2-18.5,31.2C17.2,41.5,21.2,47.1,28.2,47.1z M59.2,47.1c5.4,0,9.3-3.7,9.3-9.6c0-5.8-4.5-8.9-9.3-8.9	c-0.5,0-1,0-1.6,0.2c0-9.8,4.4-15.9,13.3-21.7l-4.2-6.8c-12.1,7-18.6,17.2-18.6,31.2C48.2,41.5,52.2,47.1,59.2,47.1z" fill="#{$color}"/></svg>';
+    @return '<svg viewBox="0 0 88 48" fill="#F08080"><path d="M28.2,47.1c5.4,0,9.3-3.7,9.3-9.6c0-5.8-4.5-8.9-9.3-8.9c-0.5,0-1.1,0-1.6,0.2C26.7,18.9,31,12.8,40,7l-4.2-6.8	c-12.1,7-18.5,17.2-18.5,31.2C17.2,41.5,21.2,47.1,28.2,47.1z M59.2,47.1c5.4,0,9.3-3.7,9.3-9.6c0-5.8-4.5-8.9-9.3-8.9	c-0.5,0-1,0-1.6,0.2c0-9.8,4.4-15.9,13.3-21.7l-4.2-6.8c-12.1,7-18.6,17.2-18.6,31.2C48.2,41.5,52.2,47.1,59.2,47.1z" fill="#{$color}"/></svg>';
```
Comment (heatherdesigns) [inline docroot/themes/humsci/humsci_basic/src/scss/tools/_functions.icons.scss]: OOPS!!!! TODO: revert change!!

**Comment by rkelbel** [comment]
This is **AMAZING** work Heather!   Soooo much awesome work to comb through. 🤩 

great work clap

**Comment by rkelbel** [comment]
**Dark Inversion feedback:**
Ah Dark Inversion pattern, we meet again..


For hb-dark-pattern styling, when this is the top item added to a component area and we use the utility class for dark inversion there is kind of a slightly awkward top line to the testimonial on Colorful.

Screen Shot 2021-02-05 at 4 50 13 PM

I wasn't sure if this was expected and we might refine on feedback if this edge case gets used or if perhaps adding a little bit of padding around the top of the quotation mark might help it feel less awkward being right against the end of the color break.  Another option perhaps is to add padding to the top of the entire testimonial so that the dark background comes up to just above the testimonial quotation mark and makes in a large dark component block rather than having the quotation breaking the top of the grid. That could be set only using it when the dark pattern or inversion is applied.  This is a very specific use and condition, so I thought I'd share if we've already okayed this then feel free to disregard this feedback.

----

Some of the color pairings are not ideal with dark inversion for accessibility (Lake and Mountain in Colorful being the biggest concerns), I believe we have some refinement work coming along with color pairings and the dark inversion background colors, is that accurate?  Just confirming so we know that this is expected when it comes to QA testing on this work. 

image

image

----

In Views though, the dark inversion setting applies and looks different, the testimonial text is white.
image

image


(Traditional seems to be consistent between the utility class and the inversion setting though)

Top is a component, bottom is the start of the view:
image

**Comment by rkelbel** [comment]
Testimonial Views Pattern:
Heather you did great work here to get this setup and the weird issues I'll list are not really a reflection of the work as much as content weirdness being filtered through the pattern.

I think this likely is not something too big for us to worry about, however, I figured I would leave some notes here that we might be able to defer to for QA testing questions or notes.

Ideally, the content being applied with this pattern should be set up to be more consistent with the content the pattern is designed to display.  This can be applied to any view with the view pattern and that's the way it's been tested. However, with that, some weirdness does come up.

If any of the fields chosen contains h2's or other heading elements, they will display at that size, this may be something we want to reset and override from happening, or the person creating the view to adjust the content to not be heading html. If this is a question of wanting heading hierarchy in any of these, we will want to apply some reset or overriding to keep the design accurate and keep heading elements if desired. Could easily be a use case refinement for later.

Position:
image

Name
image

Quote:
image

**Comment by rkelbel** [comment]
**Wells:**
The "punctuation mark" icon has a white background on it on Traditional and Colorful, when on a full-width well background it doesn't match. It's fine on the limited well size though.

image

**Well w/ hb-dark-pattern:**
The well with the component areas with utility class hb-dark-pattern causes problems with the accessibility of the text as well as the color accents and punctuation icon background color.

Limited Well, Traditional, Cardinal below (each color pairing has issues here though):
image

Full well, Colorful, Cardinal:
image

**Comment by rkelbel** [comment]
Multi-Testimonials in a row and/or a well looks awesome! 🥇 
image

**Comment by rkelbel** [comment]
The ` hb-blockquote-black-text` works and looks great, this is a fantastic modifier for this look! 🎉 
image

-----

It probably goes without saying, they should avoid using the dark inversion or `hb-dark-pattern` while also applying the  `hb-blockquote-black-text` class. 😆 😆 
image

**Comment by heatherdesigns** [comment]
@rkelbel This is ready for another look. :eyes:

We have a dark color pattern / inversion refinement card in the backlog. The colors used on the testimonial for dark patterns / inversion matches the design comp. I agree that it needs refined. I'm going to add this work to [STN-582](https://sparkbox.atlassian.net/browse/STN-582).

As for the headings elements on fields in views, I'd like to ask H&S if this is a concern before jumping in and writing styles that may not be necessary. As a brand new paragraph / UI pattern I'm curious if there are plans to use this with existing data. If not, then it should be good as is. 🤞  If so, I think that work could be addressed in a refinement card.

**Review by rkelbel — APPROVED** [review]
Awesome work Heather!
This is a great component and will look awesome on all the sites.

high five cat

**Pull request merged**